### PR TITLE
LPD-40175 Allow exporting and importing one Page referencing Asset Categories Navigation widget across environments via Headless APIs

### PR DESCRIPTION
**Pull request by monaco-fabio**
https://liferay.atlassian.net/browse/LPD-40175

### CODE REVIEW

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-asset-categories-navigation-web/src/main/java/com/liferay/commerce/product/asset/categories/navigation/web/internal/configuration/CPAssetCategoriesNavigationPortletInstanceConfiguration.java [hunk 1]
```diff
@@ -23,16 +23,49 @@
...
-	@Meta.AD(name = "asset-vocabulary-id", required = false)
+	@Meta.AD(
+		deflt = "", name = "asset-vocabulary-external-reference-code",
+		required = false
+	)
+	public String assetVocabularyExternalReferenceCode();
+
+	@Meta.AD(
+		deflt = "", description = "asset-vocabulary-id-description",
+		name = "asset-vocabulary-id-description", required = false
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-asset-categories-navigation-web/src/main/java/com/liferay/commerce/product/asset/categories/navigation/web/internal/configuration/CPAssetCategoriesNavigationPortletInstanceConfiguration.java:34]: name is wrong, it should be `asset-vocabulary-id`

File: modules/apps/commerce/commerce-product-asset-categories-navigation-web/src/main/java/com/liferay/commerce/product/asset/categories/navigation/web/internal/configuration/CPAssetCategoriesNavigationPortletInstanceConfiguration.java [hunk 2]
```diff
@@ -23,16 +23,49 @@
...
+
+	@Meta.AD(
+		deflt = "", name = "root-asset-category-external-reference-code",
+		required = false
+	)
+	public String rootAssetCategoryExternalReferenceCode();
+
+	@Meta.AD(
+		deflt = "", description = "root-asset-category-id-description",
+		name = "root-asset-category-id-description", required = false
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-asset-categories-navigation-web/src/main/java/com/liferay/commerce/product/asset/categories/navigation/web/internal/configuration/CPAssetCategoriesNavigationPortletInstanceConfiguration.java:67]: name is wrong, it should be `root-asset-category-id`

File: modules/apps/commerce/commerce-product-asset-categories-navigation-web/src/main/java/com/liferay/commerce/product/asset/categories/navigation/web/internal/display/context/CPAssetCategoriesNavigationDisplayContext.java [hunk 3]
```diff
@@ -253,9 +287,27 @@ public String getFriendlyURL(long categoryId, ThemeDisplay themeDisplay)
...
+
+		if (Validator.isNotNull(rootAssetCategoryExternalReferenceCode)) {
+			assetCategory =
+				_assetCategoryService.fetchCategoryByExternalReferenceCode(
+					rootAssetCategoryExternalReferenceCode,
+					_themeDisplay.getCompanyGroupId());
+		}
+
+		if (assetCategory != null) {
+			return String.valueOf(assetCategory.getCategoryId());
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-asset-categories-navigation-web/src/main/java/com/liferay/commerce/product/asset/categories/navigation/web/internal/display/context/CPAssetCategoriesNavigationDisplayContext.java:307]: Why is the method called `getRootAssetCategoryExternalReferenceCode` and we are returning the ID?

**Comment by monaco-fabio** [comment]
ci:report:69936183

**Comment by monaco-fabio** [comment]
ci:report:69936190

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@monaco-fabio please check my comments, thank you!

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:relevant

**Review by alessiorendina — APPROVED** [review]
LGTM, thank you!

**Comment by monaco-fabio** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:forward:force

**Pull request closed without merging**